src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -708,6 +703,11 @@ private void ResetConnections()
                 lock (_writeQueueLock) {
                     Misc.SafeSwap(ref _writeQueue, null);
                 }
+
+                NetworkStream?.Dispose();
+                NetworkStream = null;
+                _readWriteCancellationTokenSource?.Dispose();
+                _readWriteCancellationTokenSource = null;
```

Comment:
Should the following be done after waiting for the write queue as well?

```
_client?.Dispose();
_client = null;
_readWriteCancellationTokenSource?.Cancel();

_receivePause?.Dispose();
_receivePause = null;
```

Reply:
_client should be disposed asap, and _readWriteCancellationTokenSource need to call cancel soon as well, and also we don't want to receive anything at this point so dispose it should be fine. I think only things matter to wait for finishing the write queue are NetworkStream and _readWriteCancellationTokenSource..

Reply:
Sorry, I am not familiar with all the code here so I will need to ask questions.

If disposing `_client` and `_readWriteCancellationTokenSource`, can anything in the write queue be sent?

Reply:
My original thought is to stop write/sending to remote at this point since we are not going to care the responses but now I see your point. Let me make corrections... The goal is to keep the connection and finish sending all the write queue to remote before socket close.

Reply:
Here is the comment about the close function.

https://github.com/couchbase/couchbase-lite-core/blob/master/C/include/c4SocketTypes.h#L113-L120

CC: @borrrden for review

Reply:
I agree that if we are waiting for the writeQueue to be "complete" we need to empty it, but it will never be empty unless we process all the messages out of it.  It makes the wait loop on writeQueue pointless since if the queue is not empty, waiting is not going to change that.  That's the smell for me.  If we want to ignore any further writes then we don't need to wait for the writeQueue to be completed.